cmd/manager/main.go

```diff
@@ -337,6 +392,14 @@ func main() {
 
 	//+kubebuilder:scaffold:builder
 
+	if !disableMetricsTLS && certWatcher != nil {
```

Comment:
Could this be moved up to the section starting at line 188 where you're checking !disableMetricsTLS already? Near the end of it after you're creating the certWatcher, around 210 or something?

Reply:
I cannot be done before the `mgr, err := ctrl.NewManager` and must be done with the others controllers. 
We can move to up but not close to 188 line. It can be after the `certPoolWatcher ` for example,